OpenChange/MAPIStoreContactsMessage.m

```diff
@@ -1373,4 +1375,25 @@ - (void) save:(TALLOC_CTX *) memCtx
   [self updateVersions];
 }
 
+- (void) save:(TALLOC_CTX *) memCtx
+{
+  NSString *messageClass = [properties objectForKey: MAPIPropertyKey(PR_MESSAGE_CLASS_UNICODE)];
+  if (!messageClass) 
+    {
+      [self logWithFormat: @"ERROR: No message class in ContactMessage. Not saving anything"];
+      return;      
+    }
+
+  if ([messageClass isEqualToString: @"IPM.Contact"])
+    [self saveContact: memCtx]; 
```

Comment:
We are changing here the behaviour a lot without any apparent benefit if there is no PidTagMessageClass or it is a subtype of contact. I'm more inclined to have it as an `else` clause.